api/src/main/java/org/codice/ddf/admin/api/validation/SourceValidationUtils.java

```diff
@@ -71,4 +81,54 @@
         }
         return validationResults;
     }
+
+    /**
+     * Validates the {@param sourceName} against the existing source names of the configuration's for
+     * the given {@param factoryPids} using the {@param configurator}. An empty {@link List} will be returned
+     * if there are no existing source names with with name {@param sourceName}, or a {@link List} containing
+     * {@link ConfigurationMessage}s if there are errors.
+     *
+     * @param  sourceName a non null name to validate
+     * @param  factoryPids a list of non null factory pids of the configuration to validate names against
+     * @param  configurator configurator to fetch configurations for the given {@param factoryPids}
+     * @return a {@link List} of {@link ConfigurationMessage}s containing failure messages, or empty {@link List}
+     *         if there are no duplicate source names found
+     */
+    public List<ConfigurationMessage> validateSourceName(@Nonnull String sourceName,
+            @Nonnull List<String> factoryPids, Configurator configurator) {
+        if(configurator == null) {
+            configurator = new Configurator();
+        }
+
+        List<Map<String, Map<String, Object>>> configurations = factoryPids.stream()
+                .map(configurator::getManagedServiceConfigs)
+                .filter(config -> !config.isEmpty())
+                .collect(Collectors.toList());
+
+        for (Map<String, Map<String, Object>> config : configurations) {
+            for (Map<String, Object> entry : config.values()) {
+                Object id = entry.get("id");
+                if (id instanceof String) {
+                    String name = (String) id;
+                    if (StringUtils.isNotEmpty(name) && sourceName.equals(name)) {
+                        LOGGER.debug(
+                                "Found duplicate existing source name when creating source with name \"{}\"",
+                                sourceName);
+                        return Collections.singletonList(createInvalidFieldMsg(String.format(
+                                "A source with the name \"%s\" is already in use. Please choose another name.",
+                                sourceName), SOURCE_NAME));
+                    }
+                } else {
+                    LOGGER.debug(
+                            "Unable to validate duplicity of the incoming configuration's source name \"{}\"",
+                            sourceName);
+                    return Collections.singletonList(createInvalidFieldMsg(String.format(String.format(
```

Comment:
This doesn't look like an invalid field, it looks more like an internal-ish error. I'm not even sure if it's worth reporting this, it might be a good idea to remove this entire else block. The String.format within a String.format also could be cleaned up.